src/test/client.rs

```diff
@@ -663,3 +677,134 @@ async fn plain_auth() {
         }
     );
 }
+
+/// Test verifies that retrying a commitTransaction operation after a checkOut
+/// failure works.
+#[cfg_attr(feature = "tokio-runtime", tokio::test(flavor = "multi_thread"))]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn retry_commit_txn_check_out() {
```

Comment:
Hmm, I'm not sure. The reason we ran into it was because we use the cached hello response from the Connection to determine if we should append a txnId (i.e. if the server is a non-standalone). If other drivers are doing that, I can see them potentially running into this.

Reply:
Filed https://jira.mongodb.org/browse/DRIVERS-2324